core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableCorrelate.java

```diff
@@ -81,6 +87,35 @@ public static EnumerableCorrelate create(
         traitSet, left, right, correlationId, requiredColumns, joinType);
   }
 
+  @Override public Pair<RelTraitSet, List<RelTraitSet>> passThroughTraits(
+      final RelTraitSet required) {
+    final RelCollation collation = required.getCollation();
+    if (collation == null || collation == RelCollations.EMPTY) {
+      return null;
+    }
```

Comment:
Not very familiar with `EnumerableCorrelate`, will it need to deal with RIGHT OUTER/FULL OUTER JOIN style case?

Reply:
Subquery, lateral query can only be inner, left outer correlate.
I don't know how to generate full/right outer correlate.

Reply:
Correlate operator (base class) can only handle LEFT / INNER / SEMI / ANTI join. We do not need to consider RIGHT/ FULL in this case.